src/test/functionDetails.test.ts

```diff
@@ -0,0 +1,275 @@
+import { beforeAll, expect, test } from "bun:test";
+import { extractFunctionName } from "../control-flow/function-utils.ts";
+import { initParsers, iterFunctions } from "../file-parsing/vite.ts";
+
+// Initialize parsers once before all tests
+beforeAll(async () => {
+  await initParsers();
+});
+// Go Tests
+test("Go: function_declaration", () => {
+  const code = `func main() {
+    log.SetPrefix("mkzip: ")
+    log.SetFlags(0)
+  }`;
+  const funcIterator = iterFunctions(code, "Go");
+  const func = funcIterator.next().value; // Extract the first function node
+  expect(extractFunctionName(func, "Go")).toBe("main");
+});
+test("Go: method_declaration", () => {
+  const code = `func (l *Literal) Execute(writer io.Writer, data map[string]interface{}) core.ExecuteState {
+    writer.Write(l.Value)
+    return core.Normal
+  }`;
+  const funcIterator = iterFunctions(code, "Go");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "Go")).toBe("Execute");
+});
+test("Go: func_literal assigned to variable", () => {
+  const code = `isDebugInts := func(s string) bool {
+		// Some systems use leading _ to denote non-assembly symbols.
+		return s == "__cgodebug_ints" || s == "___cgodebug_ints"
+	}`;
+  const funcIterator = iterFunctions(code, "Go");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "Go")).toBe("isDebugInts");
+});
+test("Go: func_literal with no parent", () => {
+  const code = `return func(x int) int {
+        sum += x
+        return sum
+    }`;
+  const funcIterator = iterFunctions(code, "Go");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "Go")).toBe("Anonymous");
+});
+test("Go: nested function", () => {
+  const code = `func outer() {
+    func inner() {
+      fmt.Println("Inner function")
+    }
+    inner()
+  }`;
+  const funcIterator = iterFunctions(code, "Go");
+  const func = funcIterator.next().value;
+  expect(extractFunctionName(func, "Go")).toBe("outer");
+});
```

Comment:
This seems to be testing the order of function iteration, not the name extraction mechanism.